[xuelinglow] iP

xuelinglow opened the pull request:
# MichelleBot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes)) 

MichelleBot frees your mind of having to remember things you need to do. It's,

  - text-based
  - easy to learn
  - FAST SUPER FAST to use

All you need to do is,

 1.   download it from [here](https://github.com/xuelinglow/ip/releases/tag/A-Jar).
 2.  double-click it.
 3.   add your tasks.
 4.   let it manage your tasks for you 😉

And it is FREE!

Features:

- [x]  Managing tasks, deadlines, events
- [ ] Managing deadlines
- [ ] Reminders

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

xuelinglow committed:
Duke Level-0: Rename, Greet, Exit

xuelinglow committed:
Duke Level-1: Echo

xuelinglow committed:
Duke Level-2: Add, List

xuelinglow committed:
Duke Level-3: Mark as Done

xuelinglow committed:
Duke Level-4: ToDo, Event, Deadline

xuelinglow committed:
Duke A-TextUiTesting: Automated Text UI Testing

xuelinglow committed:
Duke Level-5: Handle Errors

xuelinglow committed:
Duke Level-6: Delete

xuelinglow committed:
Duke A-Enums: Use Enums

xuelinglow committed:
Duke Level-7: Save

Duke Level-7: Save

Duke Level-7: Save

Duke Level-7: Save

test

xuelinglow committed:
Duke Level-7: Save

Duke Level-7: Save

xuelinglow committed:
Merge branch 'branch-Level-7'

xuelinglow committed:
Duke Level-8: Dates and Times

xuelinglow committed:
Duke Level-8: Dates and Times

xuelinglow committed:
Merge branch 'branch-Level-8'

xuelinglow committed:
Duke Level-8: Dates and Times

xuelinglow committed:
Duke A-MoreOOP: Use More OOP

xuelinglow committed:
Duke A-Packages: Organize into Packages

xuelinglow committed:
Merge remote-tracking branch 'origin/add-gradle-support'

xuelinglow committed:
Duke A-Gradle: Use Gradle

xuelinglow committed:
Duke A-JUnit: Add JUnit Tests

xuelinglow committed:
Duke A-CodingStandard: Follow the Coding Standard

xuelinglow committed:
Duke A-CodingStandard: Follow the Coding Standard

xuelinglow committed:
Duke A-CodingStandard: Follow the Coding Standard

xuelinglow committed:
Duke A-JavaDoc: JavaDoc

xuelinglow committed:
Duke A-CodingStandard: Follow the Coding Standard

xuelinglow committed:
Duke A-JavaDoc: JavaDoc

xuelinglow committed:
Duke A-JavaDoc: JavaDoc

xuelinglow committed:
Duke Level-9: Find

zhikaiong2001 commented on src/main/java/taskList/Duke.java:
```diff
@@ -0,0 +1,37 @@
+package tasklist;
```
The name of the package seems unintuitive, something like `duke` sounds better

zhikaiong2001 commented on src/main/java/taskList/Duke.java:
```diff
@@ -0,0 +1,37 @@
+        do { // continue the program until 'bye' command is inputted.
+            String input = scanner.nextLine().trim();
+            Command c = Parser.parseCommand(input);
+            c.execute(this.taskList, this.ui, this.storage);
```
Great use of abstraction and polymorphism to handle the various commands

zhikaiong2001 commented on src/main/java/taskList/Parser.java:
```diff
@@ -0,0 +1,97 @@
+            case ADD:
+                return new AddCommand(inputList[1], 0);
+
+            case BYE:
+            case LIST:
```
Can be removed since `bye` and `list` commands with additional text following them would be handled by the default clause

zhikaiong2001 commented on src/main/java/taskList/commands/AddCommand.java:
```diff
@@ -0,0 +1,79 @@
+package tasklist.commands;
+
+import java.time.DateTimeException;
+import tasklist.*;
```
Imports should always be listed explicitly(ie. avoid wildcard)

zhikaiong2001 reviewed (commented):
Overall no big issues, though could consider adding JavaDocs for all methods/classes, good job!

YowSiaoKang commented on src/main/java/taskList/Ui.java:
```diff
@@ -0,0 +1,114 @@
+            + "list - list out the current tasks you have\n"
+            + "bye - exit the program\n"
+            + "(NOTE: deadline should be in dd-mm-yyyy hhmm format)");
+		}
```
Do take note of the Indentation here!

YowSiaoKang commented on src/main/java/taskList/Ui.java:
```diff
@@ -0,0 +1,114 @@
+ * ui.showWelcomeMessage();
+ *
+ */
+public class Ui {
```
I like how you provide a sample usage here.

YowSiaoKang commented on src/main/java/taskList/commands/ListCommand.java:
```diff
@@ -0,0 +1,21 @@
+
+public class ListCommand implements Command {
+    // public ListCommand() {
+        // }
```
Perhaps you forgot to remove these :)

YowSiaoKang commented on src/main/java/taskList/commands/ListCommand.java:
```diff
@@ -0,0 +1,21 @@
+                int itemNumber = i + 1;
+                Task listItem = taskList.getTask(i);
+                System.out.println(itemNumber + ". " + listItem);
+        }
```
Just some formatting issues here.

YowSiaoKang commented on src/main/java/taskList/TaskList.java:
```diff
@@ -0,0 +1,113 @@
+ */
+public class TaskList implements Serializable {
+	/** arraylist to store tasks */
+    private ArrayList<Task> taskList;
```
Perhaps a different variable name from the class name?

YowSiaoKang reviewed (commented):
Overall, I found your code easy to read, mostly just some formatting issues. Good Job!

xuelinglow committed:
Making some minor changes based on feedback

xuelinglow committed:
Duke A-CheckStyle: Use CheckStyle

xuelinglow committed:
rename the folders

xuelinglow committed:
Duke Level-10: GUI

xuelinglow committed:
Merge changes from branch-Level-10

xuelinglow committed:
Duke A-Assertions: Use Assertions

rickkoh commented on src/main/java/tasklist/Duke.java:
```diff
@@ -0,0 +1,49 @@
+        System.out.println(taskList.getTask(0));
+    }
+
+    // /** Function to start running the chatbot */
```
I noticed there are sections of code that are currently commented out and appear to be unused. It might be beneficial to remove these to keep the codebase clean and maintainable. If there's a specific reason they're being kept, could you please clarify?

xuelinglow committed:
Duke A-CodeQuality: Improve Code Quality

xuelinglow committed:
BCD-Extension B-ViewSchedules - implemented the feature

The current program lacks a feature to view tasks in the form of a
schedule.

It's essential to implement it as it allows users to view their tasks
and plan for them more effectively.

This is an initial implementation of the View Schedules feature to allow
users to view their deadlines and events on a calendar format.

This is done as deadlines and events consists of dates and times, so it
is easier for users to view the tasks in a calendar format.

The current version may be limited in terms of customization and
functionality, and further development is planned to expand on the
feature, provided there is enough time to do so.

xuelinglow committed:
Updated some minor changes

xuelinglow committed:
Merge branch 'branch-A-CodeQuality'

xuelinglow committed:
Update according to iP code quality feedback

Updated the code according to the iP code quality feedback given under
Issues by nus-se-script

xuelinglow committed:
Implement changes to viewSchedule, enhance GUI

Added a list at the side of the calendar so users can view their tasks
in a list format (by month).

GUI changes was also made, as well as code checkstyle changes.

xuelinglow committed:
Update user guide, helpg command

Updated user guide content to bold the commands for readability. Helpg
command was also edited to share the link to the user guide instead of a
lengthy message.

xuelinglow committed:
Update help command to link users to user guide

Updated the helpg command so that instead of a message with the details,
users are now guided to the user guide where it would easier for them to
refer to.

xuelinglow committed:
Fix some bugs in viewschedule

Fixed some bugs in viewschedule, such as the format looking weird for
some months, and users being able to see the tasks in the same month but
different year.

Some changes were also made to some commands file, as some discrepancies
were found after testing with gradlew
